go/adbc/README.md

```diff
@@ -0,0 +1,308 @@
+  regarding copyright ownership.  The ASF licenses this file
+  to you under the Apache License, Version 2.0 (the
+  "License"); you may not use this file except in compliance
+  with the License.  You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing,
+  software distributed under the License is distributed on an
+  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+  KIND, either express or implied.  See the License for the
+  specific language governing permissions and limitations
+  under the License.
+-->
+
+# adbc
+
+The `adbc` package provides Go interface definitions for using Arrow data and databases, also known as [ADBC](https://arrow.apache.org/adbc/).
+
+## Getting Started
+
+To give a brief tour of what you can do with the `adbc` package, we'll show a brief example of using the [SQLite driver](https://arrow.apache.org/adbc/current/driver/sqlite.html).
+
+### Installation
+
+First, assuming you've already created a Go module, add the `drivermgr` package to your `go.mod` by running:
+
+```sh
+go get github.com/apache/arrow-adbc/go/adbc/drivermgr
+```
+
+### Imports
+
+For imports, all we really need is `github.com/apache/arrow-adbc/go/adbc/drivermgr` to start but we'll import a few more modules now because we'll use them later.
+
+```go
+package main
+
+import (
+    "context"
+    "fmt"
+    "strings"
+
+    "github.com/apache/arrow-adbc/go/adbc"
+    "github.com/apache/arrow-adbc/go/adbc/drivermgr"
+    "github.com/apache/arrow-go/v18/arrow"
+    "github.com/apache/arrow-go/v18/arrow/array"
+    "github.com/apache/arrow-go/v18/arrow/memory"
+)
+
+func GettingStarted() error {
+    ctx := context.Background()
+
+```
+
+### Setup
+
+Any program using ADBC will start with creating a `Database`, a `Connection` to that `Database`, and usually one or more `Statement` objects.
+
+First we create a `Database`, providing `adbc_driver_sqlite` as the name of the driver to load:
+
+```go
+    var drv drivermgr.Driver
+    db, err := drv.NewDatabase(map[string]string{
+        "driver": "adbc_driver_sqlite",
+    })
+    if err != nil {
+        return err
+    }
+    defer db.Close()
+```
+
+Using the `Database` instance we created above, we can now create a `Connection` with `Open`:
+
+```go
+    conn, err := db.Open(ctx)
+    if err != nil {
+        return err
+    }
+    defer conn.Close()
+```
+
+Before we can execute any queries, we need to create a `Statement` to manage them:
+
+```go
+    stmt, err := conn.NewStatement()
+    if err != nil {
+        return err
+    }
+    defer stmt.Close()
+```
+
+Now that we have a basic setup, we'll show some of the functionality.
+
+### Executing a Query
+
+We can execute a query and get the results as Arrow data:
+
+```go
+    err = stmt.SetSqlQuery("SELECT 1, 2.0, 'Hello, world!'")
+    if err != nil {
+        return err
+    }
+
+    reader, _, err := stmt.ExecuteQuery(ctx)
```

Comment:
instead of using `_` here, should we use `n` and do something like `fmt.Println("NumRows: ", n)` to indicate what that return value is?